parsers/jscript.c

```diff
@@ -124,7 +124,8 @@ typedef enum eTokenType {
 	 * Used only in readTokenFull or lower functions. */
 	TOKEN_ATMARK,
 	TOKEN_BINARY_OPERATOR,
-	TOKEN_ARROW
+	TOKEN_ARROW,
+	TOKEN_DOTS,					/* ... */
```

Comment:
Remove trailing comma

Reply:
This is intentional.

Adding ',' at the end of the last element in a list is a good practice because a future diff adding a new element at the end of the list can be smaller; when adding TOKEN_foo, just adding a line is what we have to do like:
```diff
    TOKEN_arrow,
+   TOKEN_foo,
```
We will not have to do like:
```diff
-    TOKEN_arrow
+    TOKEN_arrow,
+    TOKEN_foo
```

With this technique, we can make the diff two lines smaller.
Generally, smaller diff helps people understand the diff.

(I'm talking about the size of a commit. I am not talking about the number of commits in a pull request.)

In my experience, this is a popular technique:
* https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/linux/netlink.h#n186
* https://sourceware.org/git/?p=glibc.git;a=blob;f=include/stdio.h;h=a6f7fd43cb664eb55f4987dfa830466bba38aa5b;hb=HEAD#l149
* https://github.com/llvm/llvm-project/blob/41fba3c107a5bc99065f3bf8b9f5b9d52eab2d98/llvm/include/llvm/BinaryFormat/ELF.h#L322

It is popular technique even in ctags:
* https://github.com/universal-ctags/ctags/blob/master/parsers/jscript.c#L97
* https://github.com/universal-ctags/ctags/blob/2052bbfcb2e06a4728417010bfa6fbb00e16a5bf/parsers/jscript.c#L254
* https://github.com/universal-ctags/ctags/blob/2052bbfcb2e06a4728417010bfa6fbb00e16a5bf/parsers/jscript.c#L281

An exception is a list we don't assume that a developer adds an element to the end of:
```
enum foo {
   A,
   B,
   COUNT_foo
};
```
This code implies a developer should add a new element before `COUNT_foo`.

A parser maintainer like you is the most valuable in this project. In most of all cases, the preferences of the parser maintainer have higher priority than the project's common coding convention. If you don't like the last ',' in the last element in a list, let me know. I will withdraw this change.

To make my intention clear, I will make a new pull request just for this change.

Reply:
See #3436.

Reply:
OK, I'll remember that for my future patches.